### PR TITLE
Import logging framework as logger

### PR DESCRIPTION
**Pull request by ligangty**

### CODE REVIEW

**Comment by ligangty** [comment]
@hjmodi Thanks for your suggestion. That's very helpful. I've imported the logging framework in this commit instead of the print function, and replace nose with pytest.
@yma88 Please check if the logger in npm part makes sense to you.

**Review by yma88 — APPROVED** [review]
make sense, thanks

File: mrrc/npm.py [hunk 1]
```diff
@@ -213,10 +216,10 @@ def gen_package_meatadata_file(version_metadata: NPMVersionMetadata, root='/'):
     version_dict[version_metadata.get_version()] = version_metadata.__dict__
     package_metadata.set_versions(version_dict)
 
-    logging(package_metadata.__str__())
+    logger.debug(package_metadata.__str__())
```
Comment (hjmodi) [inline mrrc/npm.py:219]: We shouldn't be calling __str__ function just like the __init__ function it's called by default when the object is converted to string.

File: mrrc/mrrc_logger.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+from os import fdopen, dup
+import logging
+import sys
+
+MRRC_INFO_LOGGER_FORMAT='%(asctime)s : %(message)s'
+MRRC_DEBUG_LOGGER_FORMAT='%(asctime)s : %(levelname)s - %(message)s'
+DEFAULT_LOGGER = "mrrc-uploader"
+
+class EncodedStream(object):
+    # The point of this class is to force python to enocde UTF-8
```
Comment (hjmodi) [inline mrrc/mrrc_logger.py:11]: Why do we need this?

File: mrrc/mrrc_logger.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+from locale import nl_langinfo, CODESET
+from os import fdopen, dup
+import logging
+import sys
+
+MRRC_INFO_LOGGER_FORMAT='%(asctime)s : %(message)s'
```
Comment (hjmodi) [inline mrrc/mrrc_logger.py:6]: I think it's too complicated TBH. We should simply be calling getLogger in the class to init the logger for that specific class.

File: mrrc/__init__.py [hunk 4]
```diff
@@ -1,5 +1,11 @@
 from .command import init,upload,delete,gen,ls,cli
+from .mrrc_logger import set_logging
+import logging
 
+# override this however you want
+set_logging(level=logging.INFO)
```
Comment (hjmodi) [inline mrrc/__init__.py:6]: why are we calling se_logging here and getLogger in the config.py?

File: mrrc/mrrc_logger.py (hunk 2 above)
Comment (ligangty) [reply mrrc/mrrc_logger.py:11]: I just copy-paste this from osbs as it showed in the comment that they encountered some problems during encoding to console. Maybe this is not needed, I'll remove it.

File: mrrc/__init__.py (hunk 4 above)
Comment (ligangty) [reply mrrc/__init__.py:6]: This is a universal control to init logging at info level as default. We will have some type of --debug or --verbose flag later in cli to switch to debug level.

File: mrrc/mrrc_logger.py (hunk 3 above)
Comment (ligangty) [reply mrrc/mrrc_logger.py]: You mean this mrrc_logger.py file is too complicated? Or just this format?  I think this format is universal control of the format for the "mrrc-uploader" logger which will be used in all other places. I don't think we will have any other loggers.

File: mrrc/npm.py (hunk 1 above)
Comment (ligangty) [reply mrrc/npm.py]: Removed.

**Comment by ligangty** [comment]
@hjmodi I'll merge this for following development. If any more review you can add comments in this PR and I'll check and give another pr for fixing if needed.

**Comment by hjmodi** [comment]
> @hjmodi I'll merge this for following development. If any more review you can add comments in this PR and I'll check and give another pr for fixing if needed.

@ligangty I think commenting on a closed PR will not be very helpful I'll try to update the code myself and you can tell me if any feedback. WDYT?

**Pull request merged**